code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/request/handler/GenericWhitelistedProxy.java

```diff
@@ -63,14 +64,15 @@
 import org.apache.http.impl.conn.SystemDefaultRoutePlanner;
 import org.apache.http.message.BasicNameValuePair;
 import org.apache.http.protocol.HTTP;
+import org.springframework.beans.factory.BeanNameAware;
 
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
 import com.netflix.config.DynamicBooleanProperty;
 import com.netflix.config.DynamicStringListProperty;
 
 
-public class GenericWhitelistedProxy extends AbstractResponseGenerator {
+public class GenericWhitelistedProxy extends AbstractResponseGenerator implements BeanNameAware, Named {
```

Comment:
`BeanNameAware` will make spring call setBeanName when creating this bean and `Named` will make [CollectionUtils.orderList()](https://github.com/rancher/cattle/blob/9f9fdd94034fa51c5d1dceaf0fade90d7d7ff88f/code/framework/utils/src/main/java/io/cattle/platform/util/type/CollectionUtils.java#L134) consider the name instead of just the class name.